src/Castle.Core/DynamicProxy/Generators/GeneratorUtil.cs

```diff
@@ -74,25 +74,23 @@ bool IsReadOnly(ParameterInfo parameter)
 				//
 				// The above points inform the following detection logic: First, we rely on an IL
 				// `[in]` modifier being present. This is a "fast guard" against non-`in` parameters:
-				if ((parameter.Attributes & (ParameterAttributes.In | ParameterAttributes.Out)) == ParameterAttributes.In)
-				{
-					// Here we perform the actual check. We don't rely on cmods because support
-					// for them is at current too unreliable in general, and because we wouldn't
-					// be saving much time anyway.
-					if (parameter.GetCustomAttributes(false).Any(IsIsReadOnlyAttribute))
-					{
-						return true;
-					}
-				}
+				if ((parameter.Attributes & (ParameterAttributes.In | ParameterAttributes.Out)) != ParameterAttributes.In)
+					return false;
 
-				return false;
+#if FEATURE_EMIT_CUSTOMMODIFIERS
```

Comment:
(I cannot recall exactly whether `#if FEATURE_EMIT_CUSTOMMODIFIERS` is specifically for situation where custom modifiers are *emitted*, or whether it's also necessary to guard places where one just wants to *read* them. If you've already ensured that this `#if` is necessary, then all is well. Otherwise, perhaps you could quickly check this?)

Reply:
Done, looks better. Can revert the last commit if needed.